node/router/src/lib.rs

```diff
@@ -208,9 +208,30 @@ pub trait PeerPoolHandling<N: Network>: P2P {
             .collect()
     }
 
+    /// Loads any previously cached peer addresses so they can be introduced as initial
+    /// candidate peers to connect to.
+    fn load_cached_peers(storage_mode: &StorageMode, filename: &str) -> Result<Vec<SocketAddr>> {
+        let mut peer_cache_path = aleo_ledger_dir(N::ID, storage_mode);
+        peer_cache_path.push(filename);
+
+        let peers = if let Ok(cached_peers_str) = fs::read_to_string(&peer_cache_path) {
+            let mut cached_peers = Vec::new();
+            for peer_addr_str in cached_peers_str.lines() {
+                if let Ok(addr) = SocketAddr::from_str(peer_addr_str) {
+                    cached_peers.push(addr);
+                }
+            }
+            cached_peers
+        } else {
+            Vec::new()
```

Comment:
Can we add a debug log here? Could we potentially surface e.g. filesystem permission errors with a higher level read utility?